lib/aws/lex/conversation.rb

```diff
@@ -1,12 +1,54 @@
 # frozen_string_literal: true
 
-require 'aws/lex/conversation/version'
+require_relative 'conversation/base'
 
 module Aws
   module Lex
-    module Conversation
-      class Error < StandardError; end
-      # Your code goes here...
+    class Conversation
+      include Support::Responses
+
+      attr_accessor :event, :context, :lex
+
+      def initialize(opts = {})
+        self.event = opts.fetch(:event)
+        self.context = opts.fetch(:context)
+        self.lex = Type::Event.shrink_wrap(event)
+      end
+
+      def chain
+        @chain ||= []
+      end
+
+      def handlers=(list)
+        last = nil
+
+        reversed = list.reverse.map do |element|
+          handler = element.fetch(:handler).new(
+            options: element.fetch(:options) { {} },
+            successor: last
+          )
+          last = handler
+          handler
+        end
+
+        @chain = reversed.reverse
+      end
+
+      def handlers
+        chain.map(&:class)
+      end
+
+      def respond
+        chain.first.handle(self)
+      end
+
```

Comment:
is it worth making an intent_name convenience method, or current_intent?

Reply:
Yes, that's probably a good thing! I'll add one!

Reply:
I've added `Conversation#intent_name` as a convenience method!

Reply:
Good stuff!